README.md

```diff
@@ -1,170 +1,268 @@
-- `points: number[]` : a collection of indexes of the lineString defining points between each junction
-  If there are no properties, Mapbox-gl-path create two points at the edge of the lineString and determine if the path is following direction if `directionsTheme` is defined and `isFollowingDirections` is equal at `true`.
+`layersCustomisation` parameter allows to define an infinite number of layers for each `Point` or `Linestring` in order to easily style them.
 
-Point
-
-```
-properties {
-  index: number
+```ts
+interface LayersCustomisation {
+  lineLayerList: LayerCustomisation[];
+  phantomJunctionLineLayerList: LayerCustomisation[];
+  pointLayerList: LayerCustomisation[];
 }
-```
 
-Line between points
-
-```
-properties {
-  index: number
-  isFollowingDirections: boolean
+interface LayerCustomisation {
+  id: string | undefined;
+  layout: AnyLayout;
+  paint: AnyPaint;
+  type: string | undefined;
 }
 ```
 
-phantomJunction line
+#### `lineString` - GeoJSON.Feature<LineString> | undefined
 
-```
-properties {
-  index: number
-  isPhantomJunction: boolean
-  isDeparture: boolean
-}
-```
+Like `featureCollection` parameter, it draws the lineString on the map.  
+If the lineString does not contains `path` and `point` properties to describe to `Mapbox-gl-path` how to build a feature collection, automatic Points will be added:
 
-- directionsThemes - DirectionsTheme[] | undefined
+- If the first and last point have the same coordinate, the lineString is considered to be looped so one point will be placed at that coordinate and two others will be placed at equal distance from each other.
+- Otherwise, two points should be added to the edges.
 
-```
-DirectionsTheme {
-  id: number;
-  name: string;
-  getPathByCoordinates: (
-    coordinates: number[][]
-  ) => Promise<DirectionsThemeResponse | undefined>;
-}
+If `featureCollection` parameter is set, this `lineString` parameter is ignored.
 
-DirectionsThemeResponse {
-  coordinates: number[][];
-  waypoints: Waypoints | undefined;
-}
+#### `themeSelectionType` - ThemeSelectionType | undefined
 
-Waypoints {
-  departure: number[];
-  arrival: number[];
-}
+```ts
+type ThemeSelectionType = "select" | "radioList";
 ```
 
-getPathByCoordinates function return a object of type DirectionsThemeResponse with the coordinates between two points and waypoints, if necessary, to create phantomJunction lines between waypoints and points
+Determines the UI for theme selection. Default set to `radioList`.
 
-- useRightClickToHandleActionPanel - `boolean | undefined` determine if the user should use the left or right click to handle the action panel
-- themeSelectionType - `"select" | "radioList"` UI to choose the theme. Default value is `"radioList"`
+#### `translate` - Function | undefined;
 
-### METHODS
+A function to provide locales. The default language is English (See [src/i18n.js](https://github.com/makinacorpus/mapbox-gl-path/blob/master/src/i18n.ts)).
```

Comment:
You should add description of function parameters and may be where we can found the string to translate ?

Reply:
The link to default locales is not enough? Maybe i can add them

Reply:
No no, my bad, I didn't see the link.